pt/santander.py

```diff
@@ -0,0 +1,411 @@
+            start = f"{startH:02d}:{startM:02d}"
+            finish = f"{finishH:02d}:{finishM:02d}"
+            workingHours[i] = f"{start}-{finish}"
+        else:
+            continue
+    return workingHours
+
+def frange(x, y, jump):
+    epsilon = jump * 0.1
+    while x < y + epsilon:
+        yield x
+        x += jump
+
+ALL_SERVICES = [
+    "atm",
+    "cash_out",
+    "cash_in",
+    "cheque_in",
+    "prepaid_top_up:mobile",
+    "internet_access",
+]
+
+def fetch_data():
+    pois = {}
+
+    # This can be optimized by fetching in parallel, but for simplicity we will fetch sequentially.
+    
+    # Continental Portugal
+    for lat in frange(37.0, 42.0, 0.5):
+        for lon in frange(-9.0, -6.0, 0.5):
+            print(f"Fetching data for coordinates: {lat}, {lon}")
+            data = fetch_json_data(DATA_URL, params={
+                "config": json.dumps({
+                    "coords": [
+                        round(lat, 2),
+                        round(lon, 2)
+                    ]
+                })
+            })
+            
+            for poi in data:
+                pois[poi['poicode']] = poi
+
+    # Madeira
+    for lat in frange(32.5, 33.5, 0.5):
+        for lon in frange(-17.5, -16.0, 0.5):
+            print(f"Fetching data for coordinates: {lat}, {lon}")
+            data = fetch_json_data(DATA_URL, params={
+                "config": json.dumps({
+                    "coords": [
+                        round(lat, 2),
+                        round(lon, 2)
+                    ]
+                })
+            })
+            
+            for poi in data:
+                pois[poi['poicode']] = poi
+
+    # Açores
+    for lat in frange(36.5, 40.0, 0.5):
+        for lon in frange(-31.5, -25.0, 0.5):
+            print(f"Fetching data for coordinates: {lat}, {lon}")
+            data = fetch_json_data(DATA_URL, params={
+                "config": json.dumps({
+                    "coords": [
+                        round(lat, 2),
+                        round(lon, 2)
+                    ]
+                })
+            })
+
+            for poi in data:
+                pois[poi['poicode']] = poi
+
+    results = list(pois.values())
+    results = [
+        r for r in results
+        if r["entityCode"] == "Santander_Totta"
+        and r["objectType"]["code"] == "BRANCH"
+    ]
+
+    return results
+
+if __name__ == "__main__":
+    new_data = fetch_data()
+    
+    old_data = [DiffDict(e) for e in overpass_query(
+"""
+(
+    nwr[amenity=bank][name~"Santander"](area.country);
+);
+"""
+    )]
+
+    new_node_id = -10000
+    
+    for nd in new_data:
+        public_id = nd["poicode"]
+        d = next((od for od in old_data if public_id in od[REF].split(";")), None)
+        if d is None:
+            coord = [nd["location"]["coordinates"][1], nd["location"]["coordinates"][0]]
+            ds = [x for x in old_data if not x[REF] and distance([x.lat, x.lon], coord) < 100]
```

Comment:
We shouldn't merge together branches of different types (i.e. where `subType.code` is different) solely based on their proximity, this leads to incorrect tagging down below, as well as no-op change suggestions (because tags for the same POI change back and forth during a single script run) like

<img width="333" height="245" alt="image" src="https://github.com/user-attachments/assets/5a786566-73e1-4821-ab81-c3b39c025336" />

Purely from a consumer's perspective, I'd also like to be able to tell the different POI types apart, at least to know which number to call.

Reply:
Fixed.